File: MediTrack.API/DTOs/Intake/IntakeDto.cs
using System.ComponentModel.DataAnnotations;

namespace MediTrack.API.DTOs.Intake;

public class IntakeDto
{
    [Required]
    public DateTime ScheduledAt { get; set; }
    public DateTime? TakenAt { get; set; }
    public bool IsTaken => TakenAt.HasValue;
    public string Comment { get; set; } = string.Empty;
    [Range(1, int.MaxValue, ErrorMessage = "PrescriptionId must be valid")]
    public int PrescriptionId { get; set; }
}

File: frontend/src/entities/intakes/intake.dto.ts
export interface IntakeRequestDto {
  scheduledAt: string
  takenAt: string | null  // nullable — может не принял ещё
  comment: string
  prescriptionId: number
}

export interface IntakeResponseDto {
  id: number
  scheduledAt: string
  takenAt: string | null
  isTaken: boolean
  comment: string
  medicationName: string
  doctorName: string
}